server/core/store_stats.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"math"
+	"sync"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/tikv/pd/pkg/movingaverage"
+)
+
+type storeStats struct {
+	mu       sync.RWMutex
```

Comment:
Why do we need mutex here?

Reply:
It may be read (hot scheduler) and write (store heartbeat) at the same time.

Reply:
It seems we use clone to avoid directly write?

Reply:
no. I did not clone this field.